bin/all_sky_search/pycbc_bin_trigger_rates_dq

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+""" Bin triggers by their dq value and calculate trigger rates in each bin
+"""
+import logging
+import argparse
+import pycbc
+import pycbc.events
+from pycbc.events import stat as pystat
+import numpy as np
+import h5py as h5
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+from pycbc.version import git_verbose_msg as version
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--version', action='version', version=version)
+parser.add_argument('--verbose', action="store_true")
+parser.add_argument("--ifo", type=str,required=True)
+parser.add_argument("--trig-file", required=True)
+parser.add_argument("--stat-threshold", type=float,
+                    help="Only consider triggers with statistic value "
+                    "above this threshold")
+parser.add_argument("--dq-file", required=True,nargs='+')
+parser.add_argument('--bank-file', help='hdf format template bank file',
+                    required=True)
+parser.add_argument('--background-bins', nargs='+', 
+                    help='list of background bin format strings')
+parser.add_argument("--output-file", required=True)
+parser.add_argument("--output-plot")
+parser.add_argument("--prune-number", type=int, default=0,
+                    help="Number of loudest events to remove from each split "
+                         "histogram, default 0")
+parser.add_argument("--prune-window", type=float, default=0.1,
+                    help="Time (s) to remove all triggers around a trigger "
+                         "which is loudest in each split, default 0.1s")
+
+pystat.insert_statistic_option_group(parser,
+    default_ranking_statistic='single_ranking_only')
+args = parser.parse_args()
+pycbc.init_logging(args.verbose)
+
+bank = h5.File(args.bank_file, 'r')
+
+if args.background_bins:
+    logging.info('Sorting bank into bins...')
+    data = {'mass1': bank['mass1'][:], 'mass2': bank['mass2'][:],
+            'spin1z': bank['spin1z'][:], 'spin2z': bank['spin2z'][:],
+            'f_lower': bank['f_lower']}
+    locs_dict = pycbc.events.background_bin_from_string(args.background_bins, 
+                                                        data)
+    del data
+    locs_names = [b.split(':')[0] for b in args.background_bins]
+else:
+    locs_dict = {'all_bin': np.arange(0, len(bank['mass1'][:]), 1)}
+    locs_names = ['all_bin']
+
+logging.info('Reading trigger file...')
+ifo = args.ifo
+trig_file = h5.File(args.trig_file,'r')
```

Comment:
Try to use a `with h5.File(…) as trig_file:` block here, and remove the explicit `close()` call below. Also applies to similar cases further down.

Reply:
Done